### PR TITLE
refactored mesh and sync to work with their appropriate data structures

### PR DESCRIPTION
**Pull request by antonlerner**
Now consensus is the only one using TortoiseBlock as blocks and Sync and Mesh are using on the wire format of blocks i.e struct Block.

### CODE REVIEW

File: consensus/block.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	return b.LayerIndex
+}
+
+
+
+func NewBlock(coin bool, data []byte, ts time.Time, layerId LayerID) *TortoiseBlock {
+	b := TortoiseBlock{
+		Id:         BlockID(uuid.New().ID()),
+		LayerIndex: layerId,
+		BlockVotes: make(map[BlockID]bool),
```
Comment (almogdepaz) [inline consensus/block.go:42]: vote patterns for different layers are supposed to be encoded here
so cant be map[BlockID]bool

File: mesh/block.go [hunk 2]
```diff
@@ -1,98 +1,137 @@
...
-var layerCounter LayerID = 0
-
+type Block struct {
+	Id          BlockID
+	LayerIndex  LayerID
+	Data        []byte
+	Coin        bool
+	Timestamp   time.Time
+	Txs         []SerializableTransaction
+	BlockVotes 	[]BlockID
```
Comment (almogdepaz) [inline mesh/block.go:23]: supposed to be a set of block ids for a specific layer

File: mesh/mesh.go [hunk 3]
```diff
@@ -28,44 +28,49 @@ type Mesh interface {
 	Close()
 }
 
-type mesh struct {
+type MeshValidator interface{
+	HandleIncomingLayer(layer *Layer)
+	HandleLateBlock(bl *Block)
+}
+
+type meshStore struct {
```
Comment (almogdepaz) [inline mesh/mesh.go:36]: why store ? we can just loose the interface if we don't need it

File: sync/block_listener.go [hunk 4]
```diff
@@ -50,11 +54,27 @@ func NewBlockListener(net server.Service, bv BlockValidator, layers mesh.Mesh, t
...
+	}
 	bl.RegisterMsgHandler(BLOCK, newBlockRequestHandler(layers, logger))
+
+
 	return &bl
 }
 
+func (bl *BlockListener) ListenToGossipBlocks(){
+	for{
+		select {
```
Comment (almogdepaz) [inline sync/block_listener.go:68]: should add the block to orphans (use OnNewBlock), also i would do it in the same select loop as run 
another case 
would be much easier to follow and i really don't think we need the parallelism here

File: mesh/block.go (hunk 2 above)
Comment (antonlerner) [reply mesh/block.go:23]: not from the mesh's point of view, a set is deemed not serialisable both in our internal trie and on wire. when not reading content of votes and views in the tortoise algorithm there is not a lot of benefit in keeping it as a set in opposition to a list.
The block should remain in a deterministic form

File: consensus/block.go (hunk 1 above)
Comment (antonlerner) [reply consensus/block.go:42]: this is the old impl of the tortoise, new implementation will override this probably

File: mesh/block.go (hunk 2 above)
Comment (almogdepaz) [reply mesh/block.go:23]: the main point was that we dont include partition by layer id not the set
 also im not convinced that mesh's representation of block would be the one passing on the wire

**Review by almogdepaz — APPROVED** [review]

**Pull request merged**